modules/carto/src/api/maps-v3-client.ts

```diff
@@ -137,8 +141,16 @@ function getParameters({
   const sourceName = type === MAP_TYPES.QUERY ? 'q' : 'name';
   parameters.push(encodeParameter(sourceName, source));
 
-  if (GEO_COLUMN_SUPPORT.includes(type) && geoColumn) {
-    parameters.push(encodeParameter('geo_column', geoColumn));
+  if (GEO_COLUMN_SUPPORT.includes(type)) {
+    if (geoColumn) {
```

Comment:
This method is getting quite complex. Can we move this logic into `checkFetchLayerData` and add some tests to check invalid combinations throw an error? Then this method can be simplified to just encode & append parameters when they are present

Reply:
Done, let me know if you're ok with that

Reply:
Perfect, thanks. Could you also add relevant test cases here: https://github.com/visgl/deck.gl/blob/master/test/modules/carto/api/maps-api-client.spec.js#L390-L394